hil-test/tests/uart.rs

```diff
@@ -43,6 +43,35 @@ mod tests {
         assert_eq!(byte[0], 0x42);
     }
 
+    #[test]
+    fn test_different_tolerance(mut ctx: Context) {
+        ctx.uart
+            .apply_config(
+                &uart::Config::default()
+                    .with_baudrate(19_200)
+                    .with_baudrate_tolerance(uart::BaudrateTolerance::Exact),
+            )
+            .unwrap();
+
+        ctx.uart.write_bytes(&[0x42]).unwrap();
+        let mut byte = [0u8; 1];
+        ctx.uart.read_bytes(&mut byte).unwrap();
+        assert_eq!(byte[0], 0x42);
+
+        ctx.uart
+            .apply_config(
+                &uart::Config::default()
+                    .with_baudrate(9600)
+                    .with_baudrate_tolerance(uart::BaudrateTolerance::ErrorPercent(10)),
+            )
+            .unwrap();
+
+        ctx.uart.write_bytes(&[0x42]).unwrap();
+        let mut byte = [0u8; 1];
+        ctx.uart.read_bytes(&mut byte).unwrap();
+        assert_eq!(byte[0], 0x42);
+    }
+
```

Comment:
I can't comment on the diff, but could you add some more tests to `test_send_receive_different_baud_rates_and_clock_sources` to go up to the full 5Mbaud (and maybe some intervals in between)?